### PR TITLE
Task/02 interfaces

### PR DESCRIPTION
**Pull request by DavidTereshchenko**
Acceptance criteria for 1

- [x]  Mentor is added to pr
- [x]  AC copied to pr description
- [x]  Link to deployed version is added to pr
- [x]  Interfaces for Product and Category based on [swagger] are created and exported in separate file.
- [x]  Static header and footer are created according to [mobile] and [desktop] designs.

Acceptance criteria for 1*

- [x] Shared component SocialIcon is created and used for footer icons
- [x]  Link to social network and path to icon are passed to component by the typed props

### CODE REVIEW

**Comment by tetiana-zhuchenko** [comment]
Thanks for your job. I'll try to make a review as soon as possible 🤗

**Comment by tetiana-zhuchenko** [comment]
Your layout looks pretty good, but you need to do some corrections
You need to find out how to fix this header issue

image

**Comment by tetiana-zhuchenko** [comment]
During scaling, your picture should locate in the center 

image

**Comment by tetiana-zhuchenko** [comment]
You may accomplish displaying your main page without scroll (like in design) using "vh" units in your css styles (width: 100vh)

image

Design:
image

**Comment by tetiana-zhuchenko** [comment]
Could you add the link to your page, please

image

**Comment by tetiana-zhuchenko** [comment]
You need to keep one way of naming your project folder and files structure 
you can read more in this article:
https://www.codevertiser.com/react-components-folder-structure-naming-patterns/

image

File: src/img/social/Vector.svg
Comment (tetiana-zhuchenko) [inline src/img/social/Vector.svg:1]: It's a good practice to name images in describing way. 
Ex: "instagramIcon"

Could you keep the same name pattern (camelCace, kebab-cace or other) for you images and icons?

File: src/img/social/Subtract.svg
Comment (tetiana-zhuchenko) [inline src/img/social/Subtract.svg:1]: It's a good practice to name images in describing way.
Ex: "instagramIcon"

File: src/img/mode/face.svg
Comment (tetiana-zhuchenko) [inline src/img/mode/face.svg:1]: Could you use there the full word "facebook"? Is it possible that another developer will understand in during reading your code?

File: src/img/buttons/Interface/Log_Out.svg
Comment (tetiana-zhuchenko) [inline src/img/buttons/Interface/Log_Out.svg:1]: Could you keep the same name pattern (camelCace, kebab-cace or other) for you images and icons?

File: src/img/cart.svg
Comment (tetiana-zhuchenko) [inline src/img/cart.svg:1]: Could you keep the same name pattern (camelCace, kebab-cace or other) for you images and icons?

File: src/Product.ts
Comment (tetiana-zhuchenko) [inline src/Product.ts:1]: There is the product type. Could you implement it according to the "swagger" 

image

image

Comment (tetiana-zhuchenko) [inline src/Product.ts:1]: Where is your "user" Interface?

image

Comment (tetiana-zhuchenko) [inline src/Product.ts:1]: Could you add this one?

image

image

File: src/Product.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+export interface Product {
+    id: number;
+    title: string;
+    description: string;
+    price: number;
+    images: [];
+}
+
+export interface Category {
```
Comment (tetiana-zhuchenko) [inline src/Product.ts]: This one is almost correct, good job

File: src/Product.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    id: number;
+    name: string;
+    image: string;
+    creationAt: {};
```
Comment (tetiana-zhuchenko) [inline src/Product.ts]: Could you figure out what is the type of  "creationAt" ?

image

File: src/Product.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    updatedAt: {};
```
Comment (tetiana-zhuchenko) [inline src/Product.ts]: Find out the correct type, please

File: src/Product.ts
Comment (tetiana-zhuchenko) [inline src/Product.ts:1]: P.S working with "swagger"

image

image

image

result:
image

Comment (tetiana-zhuchenko) [reply src/Product.ts:1]: It's a good practice for small apps to add one "types.ts" file in your root directory and add there all types and interfaces

File: src/components/footer/Footer.tsx
Comment (tetiana-zhuchenko) [inline src/components/footer/Footer.tsx:1]: Don't forget pleace about naming of files and folders

File: src/components/footer/Footer.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import styles from '@/components/footer/Footer.module.css';
+import SocialIcon from '@/components/socialIcon/SocialIcon.tsx';
+
+const Footer = () => {
+    const linksSocial = {
+        facebook: 'https://www.facebook.com/david.tereshchenko.77/',
+        linkedin: 'https://www.linkedin.com/in/david-tereshchenko-7bb98b240/',
+        instagram: 'https://www.instagram.com/davidushka_te/',
+    };
```
Comment (tetiana-zhuchenko) [inline src/components/footer/Footer.tsx]: Interesting idea, nice

In some cases it's better to make things simple and just add 3 separate variables. If the variables are constants, name it in constant and descriptive way (Ex: FACEBOOK_URL)

File: src/components/footer/Footer.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+    return (
+        <footer className={styles.footer}>
+            <div className={styles.footerContainer}>
+                <SocialIcon links={linksSocial} />
```
Comment (tetiana-zhuchenko) [inline src/components/footer/Footer.tsx]: Could you declare your variables in the component where you need them? If you use the variables in different components, you can use export-import.

File: src/components/header/Header.component.tsx [hunk 6]
```diff
@@ -1,2 +1,46 @@
+import login from '@/img/buttons/Interface/Log_Out.svg';
+import singUp from '@/img/buttons/User/User_Add.svg';
+import basket from '@/img/cart.svg';
+import Logo from '@/img/logo.svg';
+import divider from '@/img/mode/divider.svg';
+import moon from '@/img/mode/moon.svg';
+import whiteSun from '@/img/mode/sun.svg';
+import darkSun from '@/img/mode/sun-black.svg';
```
Comment (tetiana-zhuchenko) [inline src/components/header/Header.component.tsx]: Why you don't name your icons like there?

File: src/components/header/Header.component.tsx [hunk 7]
```diff
@@ -1,2 +1,46 @@
...
+
 import styles from './header.module.css';
-export const HeaderComponent = () => <header className={styles.header}>Header</header>;
+export const HeaderComponent = () => (
```
Comment (tetiana-zhuchenko) [inline src/components/header/Header.component.tsx]: Could you please split your component in the smaller  components?

File: src/components/header/header.module.css [hunk 8]
```diff
@@ -1,4 +1,204 @@
 .header {
```
Comment (tetiana-zhuchenko) [inline src/components/header/header.module.css]: When you'll split your header component, don't forget to split your css.

File: src/components/socialIcon/SocialIcon.tsx [hunk 9]
```diff
@@ -0,0 +1,20 @@
+import styles from '@/components/socialIcon/SocialIcon.module.css';
+import facebook from '@/img/mode/face.svg';
+import instagram from '@/img/social/Subtract.svg';
+import linkedin from '@/img/social/Vector.svg';
+
+const SocialIcon = (props: { links: { facebook: string | undefined; linkedin: string | undefined; instagram: string | undefined } }) => (
```
Comment (tetiana-zhuchenko) [inline src/components/socialIcon/SocialIcon.tsx]: You just can declare URLs there and don't use props

**Review by tetiana-zhuchenko — COMMENTED** [review]
You've made a huge work and the result is very good! You just need to add some improvements.

**Comment by tetiana-zhuchenko** [comment]
Could you keep the same name pattern (camelCace, kebab-cace or other) for you images and icons?

image

File: src/type.ts [hunk 10]
```diff
@@ -0,0 +1,56 @@
+export interface Product {
+    id: number;
+    title: string;
+    description: string;
+    price: number;
+    images: [];
+    creationAt: string;
+    updatedAt: string;
+    category: {};
```
Comment (tetiana-zhuchenko) [inline src/type.ts]: You need to define type of your category.
export interface Product {
// ...
category: Category
}

image

File: src/type.ts [hunk 11]
```diff
@@ -0,0 +1,56 @@
+export interface Product {
+    id: number;
+    title: string;
+    description: string;
+    price: number;
+    images: [];
```
Comment (tetiana-zhuchenko) [inline src/type.ts]: images: string[]

image

File: src/type.ts [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+    updatedAt: string;
+    category: {};
+}
+
+export interface ProductId {
+    id: number;
+    title: string;
+    price: number;
+    description: string;
+    images: [];
```
Comment (tetiana-zhuchenko) [inline src/type.ts]: images: string[]

File: src/type.ts [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+    creationAt: string;
+    updatedAt: string;
+    category: {};
```
Comment (tetiana-zhuchenko) [inline src/type.ts]: You need to define type of your category.
export interface Product {
// ...
category: Category
}

File: src/type.ts [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+
+export interface Category {
+    id: number;
+    name: string;
+    image: string;
+    creationAt: string;
+    updatedAt: string;
+}
+
+export interface CategoryId {
```
Comment (tetiana-zhuchenko) [inline src/type.ts]: You can reuse your types so you don't need to duplicate "export interface Category" by adding "export interface CategoryId"

File: src/img/mode/sun-black.svg
Comment (tetiana-zhuchenko) [inline src/img/mode/sun-black.svg:1]: Could you choose the one way of naming, please?

File: src/img/buttons/singUp.svg
Comment (tetiana-zhuchenko) [inline src/img/buttons/singUp.svg:1]: Could you choose the one way of naming, please?

File: src/type.ts
Comment (tetiana-zhuchenko) [inline src/type.ts:1]: The name should be in plural "types.ts" 
image

File: src/components/Header/LogoAndMode/Logo.tsx
Comment (tetiana-zhuchenko) [inline src/components/Header/LogoAndMode/Logo.tsx:1]: You don't need to nest your components. It's ok if you will have a lot of components
Move, please your "LogoAndMode" in "components" folder

image

File: src/components/Header/LogoAndMode/Logo.tsx [hunk 15]
```diff
@@ -0,0 +1,20 @@
+import styles from '@/components/Header/LogoAndMode/Logo.module.css';
+import logo from '@/img/logo.svg';
+import divider from '@/img/mode/divider.svg';
+import moon from '@/img/mode/moon.svg';
+import sun from '@/img/mode/sun.svg';
+import sunBlack from '@/img/mode/sun-black.svg';
+
+const Logo = () => (
```
Comment (tetiana-zhuchenko) [inline src/components/Header/LogoAndMode/Logo.tsx]: The name of your component should be the same as a directory where it located. And css module should have the same name too. 

image

File: src/components/Header/Menu/Menu.tsx
Comment (tetiana-zhuchenko) [inline src/components/Header/Menu/Menu.tsx:1]: You don't need to nest your components. It's ok if you will have a lot of components
Move, please your "Menu" in "components" folder

File: src/components/SocialIcon/SocialIcon.tsx [hunk 16]
```diff
@@ -0,0 +1,26 @@
+import styles from '@/components/SocialIcon/SocialIcon.module.css';
+import facebookIcon from '@/img/social/facebookIcon.svg';
+import instagramIcon from '@/img/social/instagramIcon.svg';
+import linkedinIcon from '@/img/social/linkedinIcon.svg';
+
+const SocialIcon = () => {
+    const facebookLink = 'https://www.facebook.com/david.tereshchenko.77/';
```
Comment (tetiana-zhuchenko) [inline src/components/SocialIcon/SocialIcon.tsx]: If the variables are constants, name it in constant and descriptive way (Ex: FACEBOOK_URL)

File: src/components/SocialIcon/SocialIcon.tsx [hunk 17]
```diff
@@ -0,0 +1,26 @@
...
+    const linkedinLink = 'https://www.linkedin.com/in/david-tereshchenko-7bb98b240/';
```
Comment (tetiana-zhuchenko) [inline src/components/SocialIcon/SocialIcon.tsx]: If the variables are constants, name it in constant and descriptive way (Ex: LINKEDIN_URL)

File: src/components/SocialIcon/SocialIcon.tsx [hunk 18]
```diff
@@ -0,0 +1,26 @@
...
+    const instagramLink = 'https://www.instagram.com/davidushka_te/';
```
Comment (tetiana-zhuchenko) [inline src/components/SocialIcon/SocialIcon.tsx]: If the variables are constants, name it in constant and descriptive way (Ex: INSTAGRAM_URL)

**Review by tetiana-zhuchenko — COMMENTED** [review]
Thanks for your corrections, you've done a lot! 
You just need to make some small fixes.

**Comment by DavidTereshchenko** [comment]
fixed

File: src/components/SocialIcon/SocialIcon.tsx (hunk 18 above)
Comment (DavidTereshchenko) [reply src/components/SocialIcon/SocialIcon.tsx]: fixed

File: src/components/SocialIcon/SocialIcon.tsx (hunk 16 above)
Comment (DavidTereshchenko) [reply src/components/SocialIcon/SocialIcon.tsx]: fixed

File: src/components/SocialIcon/SocialIcon.tsx (hunk 17 above)
Comment (DavidTereshchenko) [reply src/components/SocialIcon/SocialIcon.tsx]: fixed

File: src/components/Header/Menu/Menu.tsx
Comment (DavidTereshchenko) [reply src/components/Header/Menu/Menu.tsx:1]: fixed

File: src/components/Header/LogoAndMode/Logo.tsx (hunk 15 above)
Comment (DavidTereshchenko) [reply src/components/Header/LogoAndMode/Logo.tsx]: Fixed

File: src/components/Header/LogoAndMode/Logo.tsx
Comment (DavidTereshchenko) [reply src/components/Header/LogoAndMode/Logo.tsx:1]: fixed

File: src/type.ts
Comment (DavidTereshchenko) [reply src/type.ts:1]: fixed

File: src/img/buttons/singUp.svg
Comment (DavidTereshchenko) [reply src/img/buttons/singUp.svg:1]: fixed

File: src/img/mode/sun-black.svg
Comment (DavidTereshchenko) [reply src/img/mode/sun-black.svg:1]: fixed

File: src/type.ts (hunk 14 above)
Comment (DavidTereshchenko) [reply src/type.ts]: fixed

File: src/type.ts (hunk 13 above)
Comment (DavidTereshchenko) [reply src/type.ts]: fixed

File: src/type.ts (hunk 12 above)
Comment (DavidTereshchenko) [reply src/type.ts]: fixed

File: src/type.ts (hunk 11 above)
Comment (DavidTereshchenko) [reply src/type.ts]: fixed

File: src/type.ts (hunk 10 above)
Comment (DavidTereshchenko) [reply src/type.ts]: fixed

**Review by tetiana-zhuchenko — APPROVED** [review]

**Pull request merged**